transformer_engine/common/fused_attn/fused_attn.cpp

```diff
@@ -85,15 +85,26 @@ NVTE_Fused_Attn_Backend nvte_get_fused_attn_backend(
   NVTE_CHECK(q_dtype == kv_dtype, "Q and KV must have the same data type.");
   NVTE_QKV_Format qkv_format = nvte_get_qkv_format(qkv_layout);
   auto cudnn_runtime_version = cudnnGetVersion();
-  if ((q_dtype == NVTEDType::kNVTEFloat8E4M3) || (q_dtype == NVTEDType::kNVTEFloat8E5M2)
-          && (sm_arch_ >= 90)
-          && (max_seqlen_q == max_seqlen_kv)
-          && (num_attn_heads == num_gqa_groups)
-          && (max_seqlen_q <= 512)
-          && (head_dim == 64)
-          && (bias_type == NVTE_Bias_Type::NVTE_NO_BIAS)
-          && (attn_mask_type == NVTE_Mask_Type::NVTE_PADDING_MASK)
-          && (qkv_layout == NVTE_QKV_Layout::NVTE_T3HD)) {
+  if (((q_dtype == NVTEDType::kNVTEByte)
```

Comment:
Why add the Byte type here?

Reply:
We don't have the fp8 dtypes here: https://github.com/cyanguwa/TransformerEngine/blob/01685ae25f7c2834bfbfcfe3b9e50f42dc2510a6/transformer_engine/pytorch/attention.py#L3488

Reply:
Is there a way to access `fp8_meta`? DPA currently is not a `TEBaseModule`.

Reply:
There is at least `_fp8_dtype` property of the Float8Tensor you could access (not sure why we don't expose it as a regular property without `_`, @ksivaman any insight?). Then you would be able to do `query_layer._dp8_dtype` instead.

Reply:
Fixed. Using `_fp8_dtype` now.